File: sdks/node/src/modules/watermark.ts

import { Client } from "../client";

export class WatermarkClient {
  constructor(private client: Client) {}

  async embedWatermark(file: Buffer | Blob, payload: string) {
    return this.client.request("POST", "/watermark/embed", { file, payload }, true);
  }

  async verifyWatermark(file: Buffer | Blob) {
    return this.client.request("POST", "/watermark/verify", { file }, true);
  }
}

File: sdks/python/fiduscan/watermark.py

class WatermarkClient:
    def __init__(self, client):
        self.client = client

    def embed_watermark(self, file_path, payload):
        with open(file_path, "rb") as f:
            return self.client.request("POST", "/watermark/embed", files={"file": f}, data={"payload": payload})

    def verify_watermark(self, file_path):
        with open(file_path, "rb") as f:
            return self.client.request("POST", "/watermark/verify", files={"file": f})
